python/foglamp/services/core/scheduler/scheduler.py

```diff
@@ -677,11 +677,17 @@ def _schedule_first_task(self, schedule, current_time):
 
                 schedule_id = uuid.UUID(row.get('id'))
 
+                # Data from storage layer could be a string or int
+                if type(row.get('schedule_day')) is str:
```

Comment:
row.get('schedule_day') always returns a string. A better check will be:
`if row.get('schedule_day').strip().isdigit():`

Reply:
This modification is required as Postgres returns a string for 'schedule_day', say it has value 0,

then "0" is returned as per current datatype "smallint" in table definitiion.

SQLite3 returns 0, as INTEGER.

This modification is needed in order to deal with both Postgres and Sqlite